src/Components/test/E2ETest/ServerRenderingTests/EnhancedNavigationTestUtil.cs

```diff
@@ -20,16 +22,76 @@ public static void SuppressEnhancedNavigation<TServerFixture>(ServerTestBase<TSe
-                // Normally we need to navigate here first otherwise the browser isn't on the correct origin to access
-                // localStorage. But some tests are already in the right place and need to avoid extra navigation.
-                fixture.Navigate($"{fixture.ServerPathBase}/");
-                browser.Equal("Hello", () => browser.Exists(By.TagName("h1")).Text);
+                NavigateToOrigin(fixture);
+            }
+
+            try
+            {
+                ((IJavaScriptExecutor)browser).ExecuteScript("sessionStorage.length");
+            }
+            catch (Exception ex)
+            {
+                throw new InvalidOperationException("Session storage not found. Ensure that the browser is on the correct origin by navigating to a page or by setting skipNavigation to false.", ex);
             }
 
             ((IJavaScriptExecutor)browser).ExecuteScript("sessionStorage.setItem('suppress-enhanced-navigation', 'true')");
+
+            var suppressEnhancedNavigation = ((IJavaScriptExecutor)browser).ExecuteScript("return sessionStorage.getItem('suppress-enhanced-navigation');");
+            Assert.True(suppressEnhancedNavigation is not null && (string)suppressEnhancedNavigation == "true",
+                "Expected 'suppress-enhanced-navigation' to be set in sessionStorage.");
+            _isSuppressed = true;
         }
     }
 
+    public static void CleanEnhancedNavigationSuppression<TServerFixture>(ServerTestBase<TServerFixture> fixture, bool skipNavigation = false)
+        where TServerFixture : ServerFixture
+    {
+        if (!_isSuppressed)
+        {
+            return;
+        }
+
+        var browser = fixture.Browser;
+
+        try
+        {
+            // First, ensure we're on the correct origin to access sessionStorage
+            try
+            {
+                // Check if we can access sessionStorage from current location
+                ((IJavaScriptExecutor)browser).ExecuteScript("sessionStorage.length");
+            }
+            catch
+            {
+                if (skipNavigation)
+                {
+                    throw new InvalidOperationException("Session storage not found. Ensure that the browser is on the correct origin by navigating to a page or by setting skipNavigation to false.");
+                }
+                NavigateToOrigin(fixture);
+            }
+            ((IJavaScriptExecutor)browser).ExecuteScript($"sessionStorage.removeItem('suppress-enhanced-navigation')");
+        }
+        catch (WebDriverException ex) when (ex.Message.Contains("invalid session id"))
+        {
+            // Browser session is no longer valid (e.g., browser was closed)
+            // Session storage is automatically cleared when browser closes, so cleanup is already done
+            // This is expected in some tests, so we silently return
+            return;
+        }
+        finally
+        {
+            _isSuppressed = false;
+        }
+    }
```

Comment:
Rather than doing this. Can we check `Browser.SessionId == null` and just return? Another option might be `Browser.WindowHandles`

Reply:
`Browser.SessionId` does not exist. We can cast browser to driver instance but then..the session id exists even when there are no windows handlers and the browser is closed.
Referencing `WindowsHandles` or `CurrentWindowHandle` when the session is out creates the exact exception that we were catching.
I don't see a good replacement for the exception flow.